aten/src/ATen/native/SoftMax.cpp

```diff
@@ -418,6 +418,12 @@ Tensor softmax(const Tensor& input_, const int64_t dim_, c10::optional<ScalarTyp
   return result;
 }
 
+Tensor& softmax_out(const Tensor& input_, const int64_t dim_, c10::optional<ScalarType> dtype, Tensor& output_) {
+  Tensor val = at::softmax(input_, dim_, dtype);
```

Comment:
I am not sure if this achieves what we want really. Generally we need `out` variant kernels to directly write results into provided buffer instead of allocating a random buffer and then copy data over.